packages/core/contracts/oracle/implementation/DesignatedVotingV2Factory.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../common/implementation/Withdrawable.sol";
+import "./DesignatedVotingV2.sol";
+
+/**
+ * @title Factory to deploy new instances of DesignatedVotingV2 and look up previously deployed instances.
+ * @dev Allows off-chain infrastructure to look up a hot wallet's deployed DesignatedVoting contract.
+ */
+contract DesignatedVotingV2Factory is Withdrawable {
+    address private finder;
+    mapping(address => DesignatedVotingV2) public designatedVotingContracts;
+
+    /**
+     * @notice Construct the DesignatedVotingFactory contract.
+     * @param finderAddress keeps track of all contracts within the system based on their interfaceName.
+     */
+    constructor(address finderAddress) {
+        finder = finderAddress;
+    }
+
```

Comment:
Might not be appropriate for this PR, but is there anything we could add to this contract to make the migration process from the old DV contracts to the new ones easier? For instance, maybe we could have a function that permissionlessly replicates the configuration of an old DV contract in a new DV contract and adds the user to the mapping all in one go 🤷. Just something to think about.

Reply:
ye, we'll need to think more about this and can leave it for a follow on PR.